### PR TITLE
Rename CppRuntime_Clang/Gcc to CppRuntime_libcpp/libstdcpp.

### PR DESCRIPTION
**Pull request by JohanEngelen**
Also adds libcpp and libstdcpp -target= option recognition.

https://forum.dlang.org/post/dyxfxfeqvjimbvpawmdq@forum.dlang.org

### CODE REVIEW

**Comment by thewilsonator** [comment]
Please retitle the commit to "Fix bugzilla issue 24546: _description of issue_"

**Review by thewilsonator — COMMENTED** [review]
Needs a changelog entry ~~and a spec PR~~(done: https://github.com/dlang/dlang.org/pull/3852). 

Surely we should deprecate the old ones first before removing them to allow transition? and give the user notice that the old versions will now not be set.

I otherwise approve of this. Thanks!

**Comment by JohanEngelen** [comment]
> Surely we should deprecate the old ones first before removing them to allow transition? and give the user notice that the old versions will now not be set.

AFAICT there is no mechanism for deprecating version identifier names. But I've changed the code such that it still defines the old runtime names, such that it does not break any user code depending on it.

**Comment by JohanEngelen** [comment]
Probably the problem is that `d_do_test` is compiled by the host compiler, not by the just-built compiler...

File: compiler/src/dmd/cond.d [hunk 1]
```diff
@@ -659,9 +659,9 @@ extern (C++) final class VersionCondition : DVCondition
             case "AVR":
             case "BigEndian":
             case "BSD":
-            case "CppRuntime_Clang":
+            case "CppRuntime_libcpp":
             case "CppRuntime_DigitalMars":
-            case "CppRuntime_Gcc":
+            case "CppRuntime_libstdcpp":
```
Comment (ibuclaw) [inline compiler/src/dmd/cond.d:664]: These names are equally, if not more confusing as the original (ie: libcpp is the GCC C PreProcessing library).

libstdc++v3 refers to itself as (lib)stdcxx in code.

Needs a bit more thought.

**Review by ibuclaw — CHANGES_REQUESTED** [review]

File: compiler/src/dmd/cond.d (hunk 1 above)
Comment (JohanEngelen) [reply compiler/src/dmd/cond.d:664]: https://github.com/dlang/dmd/pull/16613

Comment (JohanEngelen) [reply compiler/src/dmd/cond.d:664]: thanks Ian

**Pull request merged**